data/mods/ssb/statuses.js

```diff
@@ -45,8 +45,10 @@ let BattleStatuses = {
 	},
 	acakewearingahat: {
 		noCopy: true,
-		onStart() {
+		onStart(source) {
+			source.types = ["Normal", "Ghost"];
```

Comment:
First put the chat message, then an illusion check so that snaquaza dosen't change type if it impersonates this set.

Also modify pokedex.js for type changes.

Reply:
oh, i was using OM's mon for the example